tests/specs/file.js

```diff
@@ -536,40 +810,38 @@ define(["dist/amd-resolver"], function(Resolver) {
     });
 
 
-    describe("hasProtocol", function() {
-      it("Path with forward slash", function() {
+    describe("when calling hasProtocol", function() {
+      it("a path with a forward slash has a protocol", function() {
         expect(File.hasProtocol("/test")).to.equal(true);
       });
 
-      it("Leading back slash", function() {
+      it("a path with a leading back slash has a protocol", function() {
         expect(File.hasProtocol("\\test")).to.equal(true);
       });
 
-      it("Path with leading dot", function() {
+      it("a path with a leading dot has a protocol", function() {
         expect(File.hasProtocol("./test")).to.equal(true);
       });
 
-      it("Path with 2 leading dot", function() {
+      it("a path with 2 leading dots has a protocol", function() {
         expect(File.hasProtocol("..\\test")).to.equal(true);
       });
 
-      it("Path with no leading slash or dot", function() {
+      it("a path with no leading slash or dot has a protocol", function() {
         expect(File.hasProtocol("test")).to.equal(true);
       });
 
-      it("Path with leading http protocol", function() {
+      it("a path with a leading http protocol does not have a protocol", function() {
```

Comment:
So looking at `hasProtocol` I see that my logic is inverted.  I am not sure why I did that.  So a test for `http://` should be a positive result and a test for `..\\test` should be a negative result.  So, the interface is backwards and the tests are backwards.  But `hasProtocol` only checks if a string starts with a protocol.  It does not validate where there is a valid URI or a URI at all